agent/eventhandler/task_handler.go

```diff
@@ -387,6 +410,32 @@ func (taskEvents *taskSendableEvents) sendChange(change *sendableEvent,
 		// If a send event is not already in progress, trigger the
 		// submitTaskEvents to start sending changes to ECS
 		taskEvents.sending = true
+
+		//If network connection is successfully up, keep track of how many
+		// task events are being sent in order to avoid hitting the throttlingLimit.
+		if cfg.DisconnectCapable.Enabled() && !cfg.GetDisconnectModeEnabled() {
+			if handler.taskCount == 0 {
+				logger.Debug("Starting taskCountTimer.")
```

Comment:
nit. we could add more context to this log to make it clear, such as "logger.Debug("Connection to ACS resumed. Starting task event count counter and timer to avoid SubmitTaskStateChange request throttling.")